src/components/PopupWithForm.js

```diff
@@ -3,21 +3,31 @@ import Popup from "./Popup.js";
 export default class PopupWithForm extends Popup {
   #submitHandler;
   #form;
+  #submitButton;
 
   constructor(popupSelector, submitHandler) {
     super(popupSelector);
     this.#submitHandler = submitHandler;
     this.#form = this.modalElement.querySelector(".modal__form");
+    this.#submitButton = this.#form.querySelector(".modal__save-button");
   }
 
   #getInputValues() {
-    this.inputInfo = {};
+    const inputInfo = {};
     const inputs = Array.from(this.modalElement.querySelectorAll(".modal__input"));
     inputs.forEach(input => {
-      this.inputInfo[input.placeholder] = input.value;
+      inputInfo[input.name] = input.value;
     })
 
-    return this.inputInfo;
+    return inputInfo;
+  }
+
+  changeButtonText (isLoading) {
+    if (isLoading) {
+      this.#submitButton.textContent = "Saving...";
+    } else {
+      this.#submitButton.textContent = "Save";
+    }
   }
 
   setEventListeners() {
```

Comment:
` this.close();`  in line 39

Please, delete it from line 39


You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well